cmd/kool_upgrade_available_checker.go

```diff
@@ -0,0 +1,52 @@
+package cmd
+
+import (
+	"kool-dev/kool/cmd/updater"
+
+	"time"
+)
+
+// UpdateAwareService holds functions to implement the checker to see if theres a new version available
+type UpdateAwareService struct {
+	KoolService
+
+	updater updater.Updater
+}
+
+// UpdateWrapper wraps the service with checker logic
+func UpdateWrapper(service KoolService) *UpdateAwareService {
+	return &UpdateAwareService{
+		service,
+		&updater.DefaultUpdater{RootCommand: rootCmd},
```

Comment:
We need to refactor it, so here we can receive the `DefaultUpdater` or the `FakeUpdater`